### PR TITLE
r/azurerm_application_gateway: add firewall_policy_id config to path rules

### PR DESCRIPTION
**Pull request by patst**
Fixes https://github.com/terraform-providers/terraform-provider-azurerm/issues/9040 .

Enables the configuration of `firewall_policy_id`  on path rules:

```
...
  path_rule {
        name                       = local.path_rule_name
        backend_address_pool_name  = local.backend_address_pool_name
        backend_http_settings_name = local.http_setting_name
        firewall_policy_id         = azurerm_web_application_firewall_policy.testfwp_path_rule.id
        paths = [
          "/test",
        ]
      }
...
```

The new Test passed:

```
make testacc TEST='./azurerm/internal/services/network/application_gateway_resource_test.go' TESTARGS='-run=TestAccApplicationGateway_customPathRuleFirewallPolicy' TESTTIMEOUT='120m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test ./azurerm/internal/services/network/application_gateway_resource_test.go -v -run=TestAccApplicationGateway_customPathRuleFirewallPolicy -timeout 120m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
2021/04/07 19:18:45 [DEBUG] not using binary driver name, it's no longer needed
2021/04/07 19:18:46 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccApplicationGateway_customPathRuleFirewallPolicy
=== PAUSE TestAccApplicationGateway_customPathRuleFirewallPolicy
=== CONT  TestAccApplicationGateway_customPathRuleFirewallPolicy
--- PASS: TestAccApplicationGateway_customPathRuleFirewallPolicy (1465.99s)
PASS
ok      command-line-arguments  1467.732s
```

### CODE REVIEW

File: azurerm/internal/services/network/application_gateway_resource_test.go [hunk 1]
```diff
@@ -225,6 +225,22 @@ func TestAccApplicationGateway_customHttpListenerFirewallPolicy(t *testing.T) {
...
+func TestAccApplicationGateway_customPathRuleFirewallPolicy(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_application_gateway", "test")
+	r := ApplicationGatewayResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.customPathRuleFirewallPolicy(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("url_path_map.0.path_rule.0.firewall_policy_id").Exists(),
```
Comment (tombuildsstuff) [inline azurerm/internal/services/network/application_gateway_resource_test.go:237]: the import test will check this is set into the state, so we can remove this:

```suggestion
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @patst 

Thanks for this PR.

I've taken a look through and left a couple of comments inline but on the whole this looks pretty good - if we can fix those up then this should be good to merge 👍 

Thanks!

**Comment by patst** [comment]
@tombuildsstuff thanks for the feedback, I applied your suggestions and generated the custom validator!

**Comment by patst** [comment]
@katbyte sorry for that. The validator crashed it.

I renamed it and fixed the test:

```
make testacc TEST='./azurerm/internal/services/network/application_gateway_resource_test.go' TESTARGS='-run=TestAccApplicationGateway_customPathRuleFirewallPolicy' TESTTIMEOUT='120m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test ./azurerm/internal/services/network/application_gateway_resource_test.go -v -run=TestAccApplicationGateway_customPathRuleFirewallPolicy -timeout 120m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
2021/04/09 09:38:51 [DEBUG] not using binary driver name, it's no longer needed
2021/04/09 09:38:51 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccApplicationGateway_customPathRuleFirewallPolicy
=== PAUSE TestAccApplicationGateway_customPathRuleFirewallPolicy
=== CONT  TestAccApplicationGateway_customPathRuleFirewallPolicy
--- PASS: TestAccApplicationGateway_customPathRuleFirewallPolicy (1560.17s)
PASS
ok      command-line-arguments  1561.850s

```

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Comment by tombuildsstuff** [comment]
Tests look good 👍

**Pull request merged**